addons/event/models/event_event.py

```diff
@@ -771,3 +774,107 @@ def _gc_mark_events_done(self):
         ])
         if ended_events:
             ended_events.action_set_done()
+
+    # Helpers for ESC/Label badge printing
+    def _get_event_timeframe_string(self):
+        self.ensure_one()
+        start_datetime = format_datetime(self.env, self.date_begin, self.date_tz, "short")
+        if self.is_one_day:
+            end_datetime = format_time(self.env, self.date_end, self.date_tz, "short")
+        else:
+            end_datetime = format_datetime(self.env, self.date_end, self.date_tz, "short")
+        return _("%(start_date)s to %(end_date)s", start_date=start_datetime, end_date=end_datetime)
+
+    def _print_event_template(self, layout: dict, flip=False):
+        self.ensure_one()
+        command = EscLabelCommand()
+
+        if self.badge_image:
+            command.print_image("BGFLIP" if flip else "BG", (0, layout["print_height"]) if flip else (0, 0))
+
+        print_centered_text(
+            layout,
+            text=self.name,
+            y_position=layout["event_name_y_pos"],
+            font_size=layout["event_name_font_size"],
+            command=command,
+            flip=flip
+        )
+
+        command.set_color(layout["secondary_text_color"], alpha=layout["secondary_text_alpha"])
+        print_centered_text(
+            layout,
+            text=self._get_event_timeframe_string(),
+            y_position=layout["date_y_pos"],
+            font_size=layout["details_font_size"],
+            command=command,
+            flip=flip
+        )
+
+        if self.address_id:
+            command.set_color(layout["secondary_text_color"], alpha=layout["secondary_text_alpha"])
+            print_centered_text(
+                layout,
+                text=self.address_id.name,
+                y_position=layout["address_y_pos"],
+                font_size=layout["details_font_size"],
+                command=command,
+                flip=flip
+            )
+
+        if self.company_id.logo:
+            logo_x_pos = (layout["print_width"] - layout["logo_width"]) / 2
+            logo_y_pos = layout["print_height"] * 2 - layout["logo_y_pos"] - layout["logo_height"] if flip else layout["logo_y_pos"]
+            command.print_image("LOGOFLIP" if flip else "LOGO", (logo_x_pos, logo_y_pos))
+
+        command.set_color(layout["secondary_text_color"], alpha=layout["secondary_text_alpha"])
+        print_centered_text(
+            layout,
+            text="Printed by Odoo Events App with Epson C4000e",
```

Comment:
Change from previous PR: Sponsor message directly in code (I don't like this but not sure about alternative without adding a field somewhere)

Reply:
This should be left as an empty placeholder the internal team can use to insert "Printed by Odoo Events App with Epson C4000e", otherwise all the users will have the sponsorship message